src/data/moves/move-attrs/switch-abilities-attr.ts

```diff
@@ -34,4 +37,26 @@ export class SwitchAbilitiesAttr extends MoveEffectAttr {
       !user.getAbility().hasAttrFlag(AbAttrFlag.UNSWAPPABLE_ABILITY)
       && !target.getAbility().hasAttrFlag(AbAttrFlag.UNSWAPPABLE_ABILITY);
   }
+
+  /**
+   * If the user has a {@link DETRIMENTAL_ABILITIES | detrimental ability}, or the target
+   * has one of Huge Power, Pure Power, or Contrary, grants a (+2) effect score bonus
+   */
+  override getEffectScore(user: EnemyPokemon, target: Pokemon, _move: Move): number {
+    /**
+     * Note: this only uses a subset of {@linkcode HIGH_VALUE_ABILITIES} since
+     * - Desolate Land and Primordial Sea are symmetrical effects
+     * - Wonder Guard cannot be swapped
+     */
+    const highValueSwappableAbilities = [AbilityId.HUGE_POWER, AbilityId.PURE_POWER, AbilityId.CONTRARY];
+
+    const targetHasHighValueAbility = target
+      .getAbilities({ revealedOnly: true })
+      .some((ab) => !ab.passive && highValueSwappableAbilities.includes(ab.ability.id));
+
+    if (DETRIMENTAL_ABILITIES.includes(user.getAbility().id) || targetHasHighValueAbility) {
+      return 2;
```

Comment:
This `+2` number should probably be descriptive constant value

and not a _magic number_